### ISSUE TITLE
Don't use `node_modules` directory

### ISSUE THREAD

**Issue by alexesprit** [trmcnvn/chrome-addon#9]
First of all thank you for developing this action, I found it very useful for one of my projects. But I have one suggestion.

**TL; DR**: Keeping `node_modules` in the repository is a bad thing, and I have a suggestion to get rid of it with no drawbacks. Moreover, I can do a PR to resolve this!

---

I noticed the repository contains `node_modules` directory, which contains all dependencies that this action use, including dev, which aren't required for the action users. Adding the whole `node_modules` directory to the repository is a bad idea in most cases.

Every time this action is used the action runner downloads ~16 MB (and it's ~82 MB unpacked), which is total overkill for the action.

The template `typescript-action` uses `@vercel/ncc` builder to build a dist file which is a bundle with all (excluding dev) dependencies. The generated dist file is much smaller (e.g. one of my actions is ~500 KB unpacked, though the project with `node_modules` included would require ~99 MB free space).

The only issue is to automate updating dist file on source files change. For my personal purposes I made [an action][action-update-file] that updates a given file (or multiple files) on GitHub, if it was changed by a script or another action. This action uses the same publishing model, and it could be used as a reference for updating dist files.

Note that this is just a suggestion, I do not insist that you use this approach.

What do you think?

[action-update-file]: https://github.com/marketplace/actions/update-files-on-github

**Comment by trmcnvn** [comment]
This repo pre-dates `ncc` when the official recommendation was to include `node_modules` -- seen here: https://github.com/actions/typescript-action/commit/eeed197b52c05d2538760748443cfedd69b66377

Due to this being the new official way of doing releases then I'm more than happy to accept a PR that makes this change.
